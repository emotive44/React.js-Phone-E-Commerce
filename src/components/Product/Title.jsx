import React from 'react';

const Title = ({ name, title }) => {
    if(!name || !title) { return null };
    return (
        <div className="row">
            <div className="col-10 mx-auto my-2 text-center">
                <h1 className="text-capitalize font-weight-bold">{name}</h1>
                <h3 className="text-capitalize font-weight-bold">{title}</h3>
            </div>
        </div>
    )
}

export default Title;
