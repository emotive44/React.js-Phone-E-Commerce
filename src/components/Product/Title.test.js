import React from 'react';
import Title from './Title';
import renderer from 'react-test-renderer';

describe('Title component', () => {
    test('should render undefiend if name or title is undefiend', () => {
        const component = renderer.create(<Title />);
        expect(component.toJSON()).toMatchSnapshot();
    })

    test('should render correct passed name and title', () => {
        const component = renderer.create(<Title name='some Name' title='some Title'/>);
        expect(component.toJSON()).toMatchSnapshot(`
            <div
            className="row"
            >
                <div
                    className="col-10 mx-auto my-2 text-center"
                >
                    <h1
                    className="text-capitalize font-weight-bold"
                    >
                    some Name
                    </h1>
                    <h3
                    className="text-capitalize font-weight-bold"
                    >
                    some Title
                    </h3>
                </div>
            </div>
        `);
    })
}) 