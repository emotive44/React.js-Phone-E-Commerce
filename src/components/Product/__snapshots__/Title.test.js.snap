// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render correct passed name and title: 
            <div
            className="row"
            >
                <div
                    className="col-10 mx-auto my-2 text-center"
                >
                    <h1
                    className="text-capitalize font-weight-bold"
                    >
                    some Name
                    </h1>
                    <h3
                    className="text-capitalize font-weight-bold"
                    >
                    some Title
                    </h3>
                </div>
            </div>
         1`] = `
<div
  className="row"
>
  <div
    className="col-10 mx-auto my-2 text-center"
  >
    <h1
      className="text-capitalize font-weight-bold"
    >
      some Name
    </h1>
    <h3
      className="text-capitalize font-weight-bold"
    >
      some Title
    </h3>
  </div>
</div>
`;

exports[`Title component should render undefiend if name or title is undefiend 1`] = `null`;
